src/module/entities/helper.js

```diff
@@ -6,97 +6,590 @@
-    //                     message: CONSTANTS.apiResponses.PROJECT_NOT_FOUND
-    //                 }
-    //             }
-
-    //             if (Object.keys(userRoleInformation).length > 0) {
-
-    //                 if (!projectDetails[0].userRoleInformation || !Object.keys(projectDetails[0].userRoleInformation).length > 0) {
-    //                     await projectQueries.findOneAndUpdate({
-    //                         _id: projectId
-    //                     },{
-    //                         $set: {userRoleInformation: userRoleInformation}
-    //                     })
-    //                 }
-    //             }
-
-    //             let result = await _projectInformation(projectDetails[0])
-
-    //             if (!result.success) {
-    //                 return resolve(result)
-    //             }
-
-
-    //             return resolve({
-    //                 success: true,
-    //                 message: CONSTANTS.apiResponses.PROJECT_DETAILS_FETCHED,
-    //                 data: result.data
-    //             })
-
-    //         } catch (error) {
-    //             return resolve({
-    //                 status:
-    //                     error.status ?
-    //                         error.status : HTTP_STATUS_CODE['internal_server_error'].status,
-    //                 success: false,
-    //                 message: error.message,
-    //                 data: []
-    //             })
-    //         }
-    //     })
-    // }
+						if (singleEntity.hasOwnProperty('allowedRoles')) {
+							updateData['allowedRoles'] = []
+							if (singleEntity.allowedRoles.length > 0) {
+								updateData['allowedRoles'] = await allowedRoles(singleEntity.allowedRoles)
+							}
 
+							delete singleEntity.allowedRoles
+						}
+
+						let columnsToUpdate = _.omitBy(singleEntity, (value, key) => {
+							return _.startsWith(key, '_')
+						})
+
+						Object.keys(columnsToUpdate).forEach((key) => {
+							updateData[`metaInformation.${key}`] = columnsToUpdate[key]
+						})
+
+						if (Object.keys(updateData).length > 0) {
+							let updateEntity = await database.models.entities.findOneAndUpdate(
+								{ _id: singleEntity['_SYSTEM_ID'] },
+								{ $set: updateData },
+								{ _id: 1 }
+							)
+
+							if (!updateEntity || !updateEntity._id) {
+								singleEntity['UPDATE_STATUS'] = 'Entity Not Updated'
+							} else {
+								singleEntity['UPDATE_STATUS'] = 'Success'
+							}
+						} else {
+							singleEntity['UPDATE_STATUS'] = 'No information to update.'
+						}
+						return singleEntity
+					})
+				)
+
+				if (entityUploadedData.findIndex((entity) => entity === undefined) >= 0) {
+					throw CONSTANTS.apiResponses.SOMETHING_WRONG_INSERTED_UPDATED
+				}
+
+				return resolve(entityUploadedData)
+			} catch (error) {
+				return reject(error)
+			}
+		})
+	}
+
+	static create(body, userDetails) {
+		return new Promise(async (resolve, reject) => {
+			try {
+				let entityType = body
```

Comment:
@MallanagoudaB please recheck this logic. refer https://github.com/ELEVATE-Project/samiksha-service/blob/dev/module/entities/helper.js add function